src/arrow.rs

```diff
@@ -0,0 +1,331 @@
+    ///        PreambleEpoch::Jan1958
+    ///    ) else { panic!(); };
+    ///
+    ///    // arbitrary uuid, obtain actual from spacecraft
+    ///    let uuid = Uuid::nil();
+    ///
+    ///    let second_header = ArrowSecondaryHeader::new(tc, uuid)?;
+    ///
+    ///    Ok(())
+    /// }
+    /// ```
+    pub fn new(time_code: CdsTimeCode24, uuid: Uuid) -> Result<Self, Error> {
+        let uuid = uuid.into_bytes();
+        let Ok(uuid_major) = <[u8; 8]>::try_from(&uuid[0..8]) else {
+            return Err(Error::Arrow(ArrowError::FailedToParseUuid));
+        };
+
+        let Ok(uuid_minor) = <[u8; 8]>::try_from(&uuid[8..16]) else {
+            return Err(Error::Arrow(ArrowError::FailedToParseUuid));
+        };
+
+        Ok(ArrowSecondaryHeader {
+            time_code,
+            uuid_major: u64::from_be_bytes(uuid_major),
+            uuid_minor: u64::from_be_bytes(uuid_minor),
+        })
+    }
+
+    /// Unpack a secondary header from bytes
+    pub fn decode(bytes: &[u8]) -> Result<Self, Error> {
+        let Ok(arr) = <&[u8; ARROW_SECONDARY_HEADER_LEN]>::try_from(bytes) else {
+            return Err(Error::Arrow(ArrowError::SecondaryHeaderUnpackFailed));
+        };
+
+        let Ok(header) = ArrowSecondaryHeader::unpack(arr) else {
+            return Err(Error::Arrow(ArrowError::SecondaryHeaderUnpackFailed));
+        };
+
+        Ok(header)
+    }
+
+    /// Writes an ArrowSecondaryHeader to a target buffer and return the
+    ///  number of bytes written (should be [`ARROW_SECONDARY_HEADER_LEN`]).
+    pub fn encode(&self, target: &mut [u8]) -> Result<usize, Error> {
+        let Ok(packed_header) = self.pack() else {
+            return Err(Error::Arrow(ArrowError::SecondaryHeaderPackFailed));
+        };
+
+        if packed_header.len() > target.len() {
+            return Err(Error::Arrow(ArrowError::SecondaryHeaderPackFailed));
+        }
+
+        target[..packed_header.len()].copy_from_slice(&packed_header);
+        Ok(packed_header.len())
+    }
+
+    /// Returns a reference to the time code field
+    pub fn time_code(&self) -> &CdsTimeCode24 {
+        &self.time_code
+    }
+}
+
+/// An Arrow Packet is a CcsdsPacket with a standard
+///  secondary header ([`ArrowSecondaryHeader`]).
+///
+/// The "user_data" argument will be used for the "user data" field
+///  of the data segment. It should not include a header or
+///  secondary header.
+///
+/// ```text
+/// +----------------+-----------------------------------+
+/// | HEADER SEGMENT |         DATA SEGMENT              |
+/// +----------------+------------------+----------------+
+/// | Primary Header | Secondary Header |   User Data    |
+/// |                |                  |                |
+/// | 48 Bits (6B)   |  224 Bits (28B)  | 224 Bits (28B) |
+/// +----------------+------------------+----------------+
+/// |                                                    |
+/// +----------------------------------------------------+
```

Comment:
so, 62 bytes in total? 

the protocol recommends 65536 bytes as the max. so do we have any rationale for 28 bytes?